### ISSUE TITLE
Predefined Character Classes match on unicode. Might be a Trap

### ISSUE THREAD

**Issue by ufobat** [perl6/doc#2567]
## The problem
In Regular Expressions the predifined character classes like \d \w (and others) match on unicode characters as well. This is perfectly fine when you try to parse input of humans.

For example when you try to implement some kind of protocol (that does not support unicode) you must not use \d or \w but `<[0..9]>` and  `<[a..zA..Z]>`. Doing so could be a common Trap.

Currently there is just a hint that describes this behavour:

```
/ <[\d] - [13579]> /;
# starts with \d and removes odd ASCII digits, but not quite the same as 
/ <[02468]> /;
# because the first one also contains "weird" unicodey digits 
```

## Suggestions

I think there must be warning about this trap in the regex documentation.

**Comment by JJ** [comment]
I don't quite follow you from the "input of humans" part. What you are saying is that we shouldn't use \d to match Arabic numerals, because they can match _any_ kind of numerals. Or because they can interpret raw input as an Unicode codepoint and match as a number. Is that correct?

**Comment by Altai-man** [comment]
I think the concern is that `\d` may give an impression like it is the same as `<[0123456789]>` and be used in places where second one is a must(e.g. implementing a protocol that does not allow Unicode).
What worse, `\d` works nicely creating an implication of a bug.
Personally, I would be more content with just a note about that near `\d` definition at regexes page, because `Traps` page, in my experience, is not something too useful to look things up(may be just me).

**Comment by jonathanstowe** [comment]
I agree that this should be documented in the Regex page though for ``\d``` it does say  _"\d matches a single digit (Unicode property N) "_ and similar for ```\w```  so perhaps some additional words to make the implications of that more explicit.

**Comment by JJ** [comment]
Technically, digit is Nd. I'll have to check if it matches Nl and Nd too.

**Comment by cfa** [comment]
By all means clarify on the regex page but the original issue does strike me as a trap (especially when migrating from standard regexes).

**Comment by AlexDaniel** [comment]
FWIW there's https://github.com/ronaldxs/Perl6-US-ASCII. It doesn't work on HEAD though (that's how I found it…), but it will soon.
